### ISSUE TITLE
Unable to dlopen librocsolver.so

### ISSUE THREAD

**Issue by deven-amd** [ROCmSoftwarePlatform/rocSOLVER#230]
TensorFlow needs to dynamically load the rocSOLVER lib at runtime (TF does the same for all ROCm libs).

Attempting to do so results in the following error
```
2021-03-24 14:12:48.355140: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] 
Could not load dynamic library 'librocsolver.so'; 
dlerror: /opt/rocm-4.0.1/lib/librocsolver.so: undefined symbol: _Z20rocblas_ger_templateILb0EfPKfKPffE15rocblas_status_P15_rocblas_handleiiPKT3_lPKT1_iiiSC_iiiPT2_iiii

2021-03-24 14:12:48.355196: F tensorflow/stream_executor/lib/statusor.cc:34] 
Attempting to fetch value instead of handling error Failed precondition: 
Could not load dynamic library 'librocsolver.so'; 
dlerror: /opt/rocm-4.0.1/lib/librocsolver.so: undefined symbol: _Z20rocblas_ger_templateILb0EfPKfKPffE15rocblas_status_P15_rocblas_handleiiPKT3_lPKT1_iiiSC_iiiPT2_iiii
```

You can reproduce this error using the following simple standalone program
```cpp
// main.cpp
// /opt/rocm/bin/hipcc -std=c++11 main.cpp -ldl -o test_runner
// ./test_runner
#include <dlfcn.h>
#include <iostream>

#define ROCSOLVER_LIB_NAME "/opt/rocm-4.0.1/lib/librocsolver.so"


void* open_roc_lib(const char* lib_name) {
  void* handle = dlopen(lib_name, RTLD_NOW | RTLD_LOCAL);
  if (!handle) {
    std::cout << dlerror() << "\n";
    return nullptr;
  }
  std::cout << "loaded " << lib_name << "\n";
  return handle;
}

int main() {
  void* rocsolver_handle = open_roc_lib(ROCSOLVER_LIB_NAME);
  dlclose(rocsolver_handle);
  return 0;
}
```

It seems that the rocblas library is not in the list of dependencies for rocsolver library...could that be the cause of this error?
```
root@prj47-rack-15:/root# ldd /opt/rocm-4.0.1/lib/librocsolver.so
	linux-vdso.so.1 (0x00007fffdccd9000)
	libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f28eff6e000)
	libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f28efd4f000)
	libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f28ef9b1000)
	libamdhip64.so.4 => /opt/rocm-4.0.1/lib/libamdhip64.so.4 (0x00007f28eea2b000)
	libstdc++.so.6 => /usr/lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f28ee6a2000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f28ee2b1000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f291a23f000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f28ee0ad000)
	libhsa-runtime64.so.1 => /opt/rocm-4.0.1/lib/libhsa-runtime64.so.1 (0x00007f28edc98000)
	libamd_comgr.so.1 => /opt/rocm-4.0.1/lib/libamd_comgr.so.1 (0x00007f28e7160000)
	librt.so.1 => /lib/x86_64-linux-gnu/librt.so.1 (0x00007f28e6f58000)
	libhsakmt.so.1 => /opt/rocm-4.0.1/lib/libhsakmt.so.1 (0x00007f28e6d2f000)
	libelf.so.1 => /usr/lib/x86_64-linux-gnu/libelf.so.1 (0x00007f28e6b15000)
	libtinfo.so.5 => /lib/x86_64-linux-gnu/libtinfo.so.5 (0x00007f28e68eb000)
	libnuma.so.1 => /usr/lib/x86_64-linux-gnu/libnuma.so.1 (0x00007f28e66e0000)
	libz.so.1 => /lib/x86_64-linux-gnu/libz.so.1 (0x00007f28e64c3000)
```

/cc @whchung @ekuznetsov139 @sunway513

**Comment by cgmb** [comment]
It works for me when I build it against the develop branch. I think this was fixed in 5004d4e8d2c0ec5c392f0256d10b8a1d59c2121b, so it should be resolved when ROCm 4.2 is released.

**Comment by cgmb** [comment]
I've verified that your sample program builds and runs as expected in the official ROCm 4.2.0 release.

Thanks for the bug report. Without your report, I wouldn't have known that this was an important property of the rocsolver library. The example program was also very helpful.
